lib/fastifySession.js

```diff
@@ -29,15 +29,15 @@ function fastifySession (fastify, options, next) {
     }
 
     const cookie = { ...options.cookie, ...cookieOpts }
-    decryptSession(sessionId, { ...options, cookie }, request, callback)
+    decryptSession(sessionId, { ...options, cookie }, request, true, callback)
   })
   fastify.decorateRequest('sessionStore', { getter: () => sessionStore })
   fastify.decorateRequest('session', null)
   fastify.addHook('onRequest', onRequest(options))
   fastify.addHook('onSend', onSend(options))
   next()
 
-  function decryptSession (sessionId, options, request, done) {
+  function decryptSession (sessionId, options, request, custom, done) {
```

Comment:
note that this is a breaking change, and I'd expect at least type definitions to be wrong? Should also be called out in the changelog somehow, and not just the title of the PR. v10 will be full of breaking changes, and I think migrating from v9 will get hard without some sort of upgrade guide

Reply:
Ugh, what a terrible signature. Make it an object.

Reply:
This decryptSession is not external API. So should be not breaking.

Reply:
>This decryptSession is not external API. So should be not breaking.

Ah, fair 👍 

---

I still think we need an upgrade guide, but I agree _this_ is internal API